src/Pipeline.cpp

```diff
@@ -565,25 +591,54 @@ void Pipeline::compile_jit(const Target &target_arg) {
                  << target << "\n";
         return;
     }
-
     // Clear all cached info in case there is an error.
     contents->invalidate_cache();
 
     // Infer an arguments vector
-    infer_arguments();
-
+    if (!debug_serialization.empty()) {
+        deserialized_pipe.infer_arguments();
+        // infer_arguments();
+    } else {
+        infer_arguments();
+    }
     // Don't actually use the return value - it embeds all constant
     // images and we don't want to do that when jitting. Instead
     // use the vector of parameters found to make a more complete
     // arguments list.
     vector<Argument> args;
-    for (const InferredArgument &arg : contents->inferred_args) {
-        args.push_back(arg.arg);
+    if (!debug_serialization.empty()) {
+        for (const InferredArgument &arg : deserialized_pipe.get_inferred_args()) {
+            args.push_back(arg.arg);
+        }
+        contents->inferred_args = deserialized_pipe.get_inferred_args();
+        // if (this->jit_handlers().custom_error != nullptr) {
+        //     debug(0) << "original jit handler custom error not null\n";
+        // }
+        // if (deserialized_pipe.jit_handlers().custom_error != nullptr) {
+        //     debug(0) << "deserialized jit handler custom print not null\n";
+        // }
+        // deserialized_pipe.jit_handlers() = this->jit_handlers();
+        // if (deserialized_pipe.jit_handlers().custom_error != nullptr) {
+        //     debug(0) << "after assigning it deserialized jit handler custom error not null\n";
+        // }
+    } else {
+        for (const InferredArgument &arg : contents->inferred_args) {
+            args.push_back(arg.arg);
+        }
     }
-
     Module module = compile_to_module(args, generate_function_name(), target).resolve_submodules();
     std::map<std::string, JITExtern> lowered_externs = contents->jit_externs;
-    contents->jit_cache = compile_jit_cache(module, std::move(args), contents->outputs, contents->jit_externs, target);
+    if (!debug_serialization.empty()) {
+        std::vector<Function> outputs;
+        for (const Func &f : deserialized_pipe.outputs()) {
+            outputs.push_back(f.function());
+        }
+        contents->jit_cache = deserialized_pipe.compile_jit_cache(module, std::move(args), outputs, deserialized_pipe.get_jit_externs(), target);
+        // deserialized_pipe.compile_jit_cache()  = contents->jit_cache;
```

Comment:
As always, commented out code should be removed or have a comment, even for a draft PR :-)